File: api/account.py
from http.server import BaseHTTPRequestHandler

from api._lib.auth import require_user
from api._lib.http import method_not_allowed, send_empty, send_error, send_json
from api._lib.store import delete_workspace_state


class handler(BaseHTTPRequestHandler):
    def do_DELETE(self):
        try:
            token, user = require_user(self.headers.get("Authorization"))
            delete_workspace_state(token, user["id"])
            send_json(
                self,
                200,
                {
                    "success": True,
                    "message": "Workspace data deleted.",
                },
            )
        except Exception as exc:
            send_error(self, exc)

    def do_OPTIONS(self):
        send_empty(self)

    def do_GET(self):
        method_not_allowed(self, "DELETE")

File: api/account.js
npm i -g vercelimport { requireUser } from "./_lib/auth.js";
import { deleteWorkspaceState } from "./_lib/store.js";

export default async function handler(req, res) {
  if (req.method !== "DELETE") return res.status(405).json({ error: "Method not allowed" });

  try {
    const { token, user } = await requireUser(req);
    await deleteWorkspaceState(token, user.id);
    res.status(200).json({ success: true, message: "Workspace data deleted." });
  } catch (err) {
    res.status(err.status || 500).json({ error: err.message });
  }
}